packages/clay-form/src/LocalizedInput.tsx

```diff
@@ -3,144 +3,170 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import ClayForm, {ClayInput} from '.';
 import {ClayButtonWithIcon} from '@clayui/button';
-const spritemap = require('@clayui/css/lib/images/icons/icons.svg');
-import ClayDropDown, {Align} from '@clayui/drop-down';
+import ClayDropDown from '@clayui/drop-down';
 import ClayIcon from '@clayui/icon';
 import ClayLabel from '@clayui/label';
 import React from 'react';
 
-const DropDownWithState: React.FunctionComponent<any> = ({
-	children,
-	language,
-	...others
-}) => {
-	const [active, setActive] = React.useState(false);
-
-	return (
-		<ClayDropDown
-			{...others}
-			active={active}
-			alignmentPosition={Align.BottomLeft}
-			onActiveChange={newVal => setActive(newVal)}
-			trigger={
-				<ClayButtonWithIcon
-					aria-expanded="false"
-					aria-haspopup="true"
-					displayType="unstyled"
-					onClick={() => setActive(!active)}
-					spritemap={spritemap}
-					symbol={language.symbol}
-					title="Open Language Dropdown"
-				/>
-			}
-		>
-			{children}
-		</ClayDropDown>
-	);
-};
+import ClayForm from './Form';
+import ClayInput from './Input';
+
+interface IItem {
+	label: string;
+	symbol: string;
+}
+
+interface ITranslations {
+	[key: string]: string;
+}
+
 interface IProps extends React.InputHTMLAttributes<HTMLInputElement> {
-	defaultLanguage: {label: string; symbol: string; value: string};
-	languages: any;
-	label?: string;
-	onInputValueChange?: any;
-	onLanguageChange?: any;
-	placeholder?: string;
-	selectedLanguage: {label: string; symbol: string; value: string};
+	ariaLabels?: {
+		default: string;
+		openLanguageDropdown: string;
+		translated: string;
+		untranslated: string;
+	};
+	helpText?: React.ReactText;
+	label?: React.ReactText;
+	prependContent?: React.ReactText;
+	locales: Array<IItem>;
+	onSelectedLocaleChange: (val: IItem) => void;
+	onTranslationsChange: (val: ITranslations) => void;
+	resultFormatter?: (val: string) => React.ReactNode;
+	selectedLocale: IItem;
+	spritemap?: string;
 	textarea?: boolean;
-	url?: boolean;
-	value?: string;
+	translations: ITranslations;
 }
 
 const ClayLocalizedInput = React.forwardRef<HTMLInputElement, IProps>(
-	({
-		defaultLanguage,
-		label = 'Check for translations',
-		languages,
-		onInputValueChange,
-		onLanguageChange,
-		placeholder = 'Text to translate...',
-		selectedLanguage,
-		url,
-		value,
-	}) => (
-		<ClayForm.Group>
-			<label htmlFor="localizedInput">{label}</label>
-
-			<ClayInput.Group>
-				{url && (
-					<ClayInput.GroupItem prepend shrink>
-						<ClayInput.GroupText>{'/'}</ClayInput.GroupText>
+	(
+		{
+			ariaLabels = {
+				default: 'Default',
+				openLanguageDropdown: 'Open Language Dropdown',
+				translated: 'Translated',
+				untranslated: 'Unstranslated',
```

Comment:
Is this the approach we'll have going forward with components that require aria-labels?
I noticed while doing the accessibility that some components use this approach.

Reply:
Yeah this is a common pattern for us to add since we need to add the ability for localization. Date picker, time picker, and color picker all use it right now I believe